src/upload.ts

```diff
@@ -55,6 +60,7 @@ const PORTAL_DIRECTORY_FILE_FIELD_NAME = "files[]";
  * @property [largeFileSize=41943040] - The size at which files are considered "large" and will be uploaded using the tus resumable upload protocol. This is the size of one chunk by default (40 mib). Note that this does not affect the actual size of chunks used by the protocol.
  * @property [errorPages] - Defines a mapping of error codes and subfiles which are to be served in case we are serving the respective error code. All subfiles referred like this must be defined with absolute paths and must exist.
  * @property [numParallelUploads=2] - Used to override the default number of parallel uploads. Disable parallel uploads by setting to 1. Note that each parallel upload must be chunk-aligned so the number of parallel uploads may be limited if some parts would end up empty.
+ * @property [staggerPercent] - A percentage from 0-100. `numParallelUploads` must be more than 1. When set, each chunk is staggered, one after another, instead of all uploads running simultaneously. The stagger percentage is how much of each chunk upload should be finished before the next chunk upload is initiated. Pass `null` to disable staggering.
```

Comment:
F/U: I don't think docstring is correct. If you pass `null` you'll hit this
```js
throw new Error(`Expected 'staggerPercent' option to be between 0 and 100, was '${opts.staggerPercent}`);
```
It seems that if you want to disable staggering, you need to pass 0.

Reply:
Don't we guard against that in the 3rd line here?

```ts
if (
  opts.staggerPercent !== undefined &&
  opts.staggerPercent !== null &&
  (opts.staggerPercent < 0 || opts.staggerPercent > 100)
) {
  throw new Error(`Expected 'staggerPercent' option to be between 0 and 100, was '${opts.staggerPercent}`);
}
```

Reply:
Maybe I misunderstood the code. Is throwing this error our way of disabling staggering? It seemed like it would fail the upload with a 400. When I read "Pass `null` to disable staggering." I imagined that the upload will succeed and the chunks will go all together, i.e. 0% staggering. 

There also seem to be no tests for this and it would be nice to cover it.

Reply:
The error is not thrown if `staggerPercent` is `null`. If it is `null` then we just don't stagger, i.e. do a regular parallel upload. I have just tested this to verify this behavior.

> There also seem to be no tests for this and it would be nice to cover it.

Unfortunately, setting up integration tests for tus is a nontrivial task. Looks like there is already an issue to track it; I added some detail:

https://linear.app/skynetlabs/issue/SKY-304/add-tus-testing-to-integration-tests-health-checks

Unit tests are also nontrivial, because there are many requests that would have to be precisely mocked to emulate the protocol.

Reply:
Oh, boy. I just fail at reading again. Of course it doesn't throw when it's a `null` - the comparison is `!==` and not `===`.